crd-generator/api/src/main/java/io/fabric8/crd/generator/CustomResourceInfo.java

```diff
@@ -121,43 +121,29 @@ public TypeDef definition() {
     return definition;
   }
 
-  private final static String TYPE_NAME = CustomResource.class.getTypeName();
-  private final static String VOID_TYPE_NAME = Void.class.getTypeName();
-
-
   public static CustomResourceInfo fromClass(Class<? extends CustomResource> customResource) {
     try {
       final CustomResource instance = customResource.getDeclaredConstructor().newInstance();
 
       final String[] shortNames = CustomResource.getShortNames(customResource);
 
-      final Scope scope = Arrays.stream(customResource.getInterfaces())
-        .filter(t -> t.getTypeName().equals(Namespaced.class.getTypeName()))
-        .findFirst().map(t -> Scope.NAMESPACED).orElse(Scope.CLUSTER);
-
       final TypeDef definition = ClassTo.TYPEDEF.apply(customResource);
-      
-      // walk the type hierarchy until we reach CustomResource or a ParameterizedType
-      Type genericSuperclass = customResource.getGenericSuperclass();
-      String typeName = genericSuperclass.getTypeName();
-      while (!TYPE_NAME.equals(typeName) && !(genericSuperclass instanceof ParameterizedType)) {
-        genericSuperclass = ((Class) genericSuperclass).getGenericSuperclass();
-        typeName = genericSuperclass.getTypeName();
+      if (DESCRIBE_TYPE_DEFS) {
```

Comment:
I'm not following the logic here (this will never be true). Is this flag supposed to be something that will be removed in the future?

Reply:
This is a flag to activate the type description. The idea is that you need to recompile so that it can be optimized away by the compiler when not active.

Reply:
Yes, but how is the flag activated, it's a `false` constant

Reply:
You recompile the code to activate it :)
Being a constant means that if the constant is false, the compiler will (presumably) remove the code during compilation.

Reply:
🤯 OK, I still don't understand how this class being part of a provided API would ever be recompiled, but I trust you know what you're doing.